config/samples/openstackclient/README.md

```diff
@@ -0,0 +1,140 @@
+# OPENSTACKCLIENT
+
+The `openstackclient` is used to interact with an existing (even minimal)
+control plane, and it can be used by `kuttl` tests to execute scripts that are
+supposed to verify that a given functionality works as expected.
+It assumes that `keystone` is available and the configmap containing cloud
+credentials already exist and can be mounted to the target `Pod`.
+
+## Deploy the openstackclient via kustomize
+
+It is possible to inject scripts providing the right mounts to the
+`openstackclient` via [kustomize](https://kustomize.io).
+In general, there are two allowed/supported scenarios:
+- inject a particular script to test a feature and verify there are no regressions
+- inject the script directory that contains the whole set of scripts: an external
+  test suite will select the right one as needed
+
+In case you want to inject a particular script, add the following snippet in the
+kustomization.yaml file:
+
+```
+- op: add
+  path: /spec/volumes/0
+  value:
+   configMap:
+     name: openstack-scripts
+     defaultMode: 755
+   name: openstack-scripts
+- op: add
+  path: /spec/containers/0/volumeMounts/0
+  value:
+    mountPath: /home/cloud-admin/import-image-container.sh
+    name: distributed-image-import
+    subPath: import-image-container.sh
+```
+
+The snippet above assumes the goal is to inject a single script only called
+`import-image-container.sh` to the `cloud-admin` `$HOME` directory, and it will
+be available under the defined `subPath`. If the purpose is to test multiple
+features, it is possible to inject the entire `scripts` directory and select
+the script that should be executed from an external test suite:
+
+```
+- op: add
+  path: /spec/containers/0/volumeMounts/0
+  value:
+   mountPath: /home/cloud-admin/scripts
+   name: openstack-scripts
+- op: add
+  path: /spec/volumes/0
+  value:
+    configMap:
+      defaultMode: 0755
+      name: openstack-scripts
+    name: openstack-scripts
+```
+
+Without a `subPath`, we can mount all the scripts provided by the `ConfigMap`
+within a `scripts` directory.
+This represents the default approach for kuttl tests.
+All the potential changes (`add/remove/replace` fields) are managed by
+`kustomize`, and the `openstackclient.yaml` definition should not be changed.
+Once everything is ready, the `openstackclient` `Pod` can be deployed with the
+following command:
+
+```bash
+oc -n <namespace> kustomize --load-restrictor LoadRestrictionsNone $pwd/openstackclient | oc apply -f -
+```
+
+**Note:**
+
+All the theory describe above assumes the `scripts` directory is used as source
+to build a `ConfigMap` that is then mounted to the `Pod`. **Before** going
+through the described steps, make sure a `ConfigMap` containing the scripts
+directory exists, or at leat each script entry is mentioned in kustomization.yaml.
```

Comment:
nit: `s/leat/least`

Reply:
thank you, fixed in the last PS.
Note that this change is connected with a follow up that I still have to propose  [1], and it establishes an easy way to help QEs to build a set of kuttl tests that are supposed to test the service functionalities.

[1] https://github.com/fmount/glance-operator/commit/db5a16bd3ea0c2d19e505b8efcfb6ff1b04b67a5